src/main/java/com/google/api/gax/grpc/RetryingCallable.java

```diff
@@ -236,19 +100,61 @@ private static CallContext getCallContextWithDeadlineAfter(
     return newCtx;
   }
 
-  private static boolean canRetry(Throwable throwable) {
-    if (!(throwable instanceof ApiException)) {
-      return false;
+  private static class GrpcRetryCallable<RequestT, ResponseT> implements Callable<ResponseT> {
+    private final FutureCallable<RequestT, ResponseT> callable;
+    private final RequestT request;
+
+    private RetryFuture<ResponseT> externalFuture;
+    private CallContext callContext;
+
+    private GrpcRetryCallable(
+        FutureCallable<RequestT, ResponseT> callable, RequestT request, CallContext callContext) {
+      this.callable = callable;
+      this.request = request;
+      this.callContext = callContext;
+    }
+
+    private void setExternalFuture(RetryFuture<ResponseT> externalFuture) {
+      this.externalFuture = externalFuture;
+    }
+
+    @Override
+    public ResponseT call() {
+      callContext =
+          getCallContextWithDeadlineAfter(
+              callContext, externalFuture.getAttemptSettings().getRpcTimeout());
+      ApiFuture<ResponseT> internalFuture = callable.futureCall(request, callContext);
+      externalFuture.setAttemptFuture(new ApiFutureToListenableFuture<>(internalFuture));
+      return null;
     }
-    ApiException apiException = (ApiException) throwable;
-    return apiException.isRetryable();
   }
 
-  private static boolean isDeadlineExceeded(Throwable throwable) {
-    if (!(throwable instanceof ApiException)) {
-      return false;
+  private static class GrpcRetryHandler<ResponseT> extends ScheduledRetryHandler<ResponseT> {
+    private GrpcRetryHandler(NanoClock clock, ScheduledExecutorService scheduler) {
+      super(clock, scheduler);
+    }
+
+    @Override
+    public boolean accept(Throwable e, RetryAttemptSettings nextAttemptSettings) {
+      return super.accept(e, nextAttemptSettings)
+          && (e instanceof ApiException)
+          && ((ApiException) e).isRetryable();
+    }
+
+    @Override
+    public RetryAttemptSettings createNextAttemptSettings(
+        Throwable e, RetryAttemptSettings prevSettings) {
+      if (((ApiException) e).getStatusCode() == Code.DEADLINE_EXCEEDED) {
```

Comment:
Yes, because otherwise it would be rejected by the accept() method of same class earlier (not making the check for the second time keeps things in a fail-fast mode).